### PR TITLE
Fix cpu profiler with -sMODULARIZE=1

### LINKED ISSUE #20425: ReferenceError: emscriptenCpuProfiler is not defined
When I link with `--cpuprofiler`, the graph shows activity and it is very helpful. Although, when I click on any button, the console shows the error:
```
Uncaught ReferenceError: emscriptenCpuProfiler is not defined     at HTMLElement.onclick (index.html:21:9)
```
Any idea how to resolve this?

My cmake link options:
```cmake
  "-lembind"
  "-sMODULARIZE=1"
  "-sEXPORT_ES6=1"
  "-sENVIRONMENT=web"
  "-sALLOW_MEMORY_GROWTH=1"
  "-sMAXIMUM_MEMORY=4GB"
  "-sSINGLE_FILE=1"
  "--memoryprofiler"
  "--cpuprofiler"
  "-sEXPORTED_FUNCTIONS=['_malloc', '_free']"
  "-sEXPORTED_RUNTIME_METHODS=['ENV']"
```

### PR DESCRIPTION
**Pull request by jspanchu**
- Fix typo in includes function
- Add missing makeXRCompatible to list of functions
- Handle injection for more webgl2 methods with multiple number of args
- Run test_cpuprofiler_memoryprofiler also with -sMODULARIZE

Fixes #20425

### CODE REVIEW

**Review by sbc100 — COMMENTED** [review]
Could you add to one of the existing test for cpu/memprofiler?  Perhaps add a `@paramaterize` to `test_cpuprofiler_memoryprofiler` in `test_browser.py` such that it also run with `-sMODULARIZE`?

**Review by sbc100 — APPROVED** [review]
Thanks for working on this!

**Review by sbc100 — APPROVED** [review]

**Comment by sbc100** [comment]
Looks like there is failure there related to closure.  `other.test_memoryprofiler_closure` is failing with: 

```
building:ERROR: /tmp/emtest_i6sbcvxy/emscripten_temp__jjxii_y/a.out.jso1.js:300:41: ERROR - [JSC_UNDEFINED_VARIABLE] variable emscriptenMemoryProfiler is undeclared
  300|   if (bytes >= 1e3 * 1024 * 1024) return emscriptenMemoryProfiler.truncDec(bytes / (1024 * 1024 * 1024)) + " GB"; else if (bytes >= 1e3 * 1024) return emscriptenMemoryProfiler.truncDec(bytes / (1024 * 1024)) + " MB"; else if (bytes >= 1e3) return emscriptenMemoryProfiler.truncDec(bytes / 1024) + " KB"; else return emscriptenMemoryProfiler.truncDec(bytes) + " B";
  ```

**Comment by jspanchu** [comment]
Should I use something other than `globalThis` or use an annotation for the function which has that line?

**Comment by sbc100** [comment]
> Should I use something other than `globalThis` or use an annotation for the function which has that line?

I don't think either of those should be needed in theory.     Can you verify that declaration of `emscriptenMemoryProfiler` comes before that line (line 300 in the output)?

**Comment by jspanchu** [comment]
This is what the near vicinity of line 300 looks like with `EMCC_DEBUG=2` and `-g1`
image

**Comment by sbc100** [comment]
> This is what the near vicinity of line 300 looks like with `EMCC_DEBUG=2` and `-g1` image

I guess you could try replacing `emscriptenMemoryProfiler` with `self` in those methods.. looking at the existing code its not clear to my why self is used in some places on not others.

**Comment by jspanchu** [comment]
It says `self` is not defined. Not clear why most of the code here uses `var self = emscriptenMemoryProfiler;` instead of `this`

**Comment by sbc100** [comment]
> var self = emscriptenMemoryProfiler;

Does `var self = emscriptenMemoryProfiler;` work for you ?

What we should probably do here is just use an ES6 class here... but that seems like a much bigger change.

How about this pattern:

```
globalThis.emscriptenMemoryProfiler = {};
Object.assign(emscriptenMemoryProfiler, {
  ...
}
```

**Comment by jspanchu** [comment]
`var self = emscriptenMemoryProfiler;` still gives an undefined variable error with closure compiler.

> How about this pattern:
> 
> globalThis.emscriptenMemoryProfiler = {};
> Object.assign(emscriptenMemoryProfiler, {
>   ...
> });

It still thinks `emscriptenMemoryProfiler` is undefined.

```
/tmp/emscripten_temp_8ibbqn15/a.out.jso1.js:278:14: ERROR - [JSC_UNDEFINED_VARIABLE] variable emscriptenMemoryProfiler is undeclared
  278| Object.assign(emscriptenMemoryProfiler, {
```

**Comment by sbc100** [comment]
> `var self = emscriptenMemoryProfiler;` still gives an undefined variable error with closure compiler.
> 
> > How about this pattern:
> > globalThis.emscriptenMemoryProfiler = {};
> > Object.assign(emscriptenMemoryProfiler, {
> > ...
> > });
> 
> It still thinks `emscriptenMemoryProfiler` is undefined.
> 
> ```
> /tmp/emscripten_temp_8ibbqn15/a.out.jso1.js:278:14: ERROR - [JSC_UNDEFINED_VARIABLE] variable emscriptenMemoryProfiler is undeclared
>   278| Object.assign(emscriptenMemoryProfiler, {
> ```

How about assigning to both `var emscriptenMemoryProfiler` (to make closure happy).. *and* `globalThis.emscriptenMemoryProfiler`? (to make the callback happy)

File: src/memoryprofiler.js [hunk 1]
```diff
@@ -90,10 +89,11 @@ globalThis.emscriptenMemoryProfiler = {
...
   formatBytes(bytes) {
-    if (bytes >= 1000*1024*1024) return emscriptenMemoryProfiler.truncDec(bytes/(1024*1024*1024)) + ' GB';
-    else if (bytes >= 1000*1024) return emscriptenMemoryProfiler.truncDec(bytes/(1024*1024)) + ' MB';
-    else if (bytes >= 1000) return emscriptenMemoryProfiler.truncDec(bytes/1024) + ' KB';
-    else return emscriptenMemoryProfiler.truncDec(bytes) + ' B';
+    var self = emscriptenMemoryProfiler;
+    if (bytes >= 1000*1024*1024) return self.truncDec(bytes/(1024*1024*1024)) + ' GB';
+    else if (bytes >= 1000*1024) return self.truncDec(bytes/(1024*1024)) + ' MB';
+    else if (bytes >= 1000) return self.truncDec(bytes/1024) + ' KB';
+    else return self.truncDec(bytes) + ' B';
```
Comment (sbc100) [inline src/memoryprofiler.js:96]: Maybe revert this part?

Comment (jspanchu) [reply src/memoryprofiler.js]: Done

**Review by sbc100 — APPROVED** [review]

**Comment by kripken** [comment]
Looks like circleCI issues are finally fixed, landing.

**Pull request merged**